packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/event_client.dart

```diff
@@ -172,35 +174,38 @@ class EventClient implements Closeable {
           'putEvents - no EventsItemResponse response received from Pinpoint',
         );
       }
-      eventsItemResponse?.forEach((eventID, eventItemResponse) {
+      eventsItemResponse?.forEach((eventId, eventItemResponse) {
         if (!_equalsIgnoreCase(eventItemResponse.message ?? '', 'Accepted')) {
           _logger.warn(
-            'putEvents - issue with eventId: $eventID \n $eventItemResponse',
+            'putEvents - issue with eventId: $eventId \n $eventItemResponse',
           );
 
-          if (_isRetryable(eventItemResponse.message ?? '')) {
+          if (_isRetryable(eventItemResponse.statusCode)) {
             _logger.warn(
-              'putEvents - recoverable issue, will attempt to resend: $eventID in next FlushEvents',
+              'putEvents - recoverable issue, will attempt to resend: $eventId in next FlushEvents',
             );
-            eventsToDelete.remove(eventID);
+
+            if (_shouldEventRetry(eventId)) {
+              eventsToDelete.remove(eventId);
+            }
           }
         }
       });
-    } on BadRequestException catch (e) {
-      _handleRecoverableException(e, eventsToDelete);
-    } on InternalServerErrorException catch (e) {
-      _handleRecoverableException(e, eventsToDelete);
-    } on NotFoundException catch (e) {
-      _handleRecoverableException(e, eventsToDelete);
-    } on TooManyRequestsException catch (e) {
-      _handleRecoverableException(e, eventsToDelete);
     } on PayloadTooLargeException catch (e) {
       _logger
         ..warn('putEvents - exception encountered: ', e)
         ..warn(
             'Pinpoint event batch limits exceeded: 100 events / 4 mb total size / 1 mb max size per event \n '
             'Reduce your event size or change number of events in a batch')
         ..warn('Unrecoverable issue, deleting cache for local event batch');
+    } on SmithyHttpException catch (e) {
+      if (e.statusCode != null && _isRetryable(e.statusCode!)) {
+        _handleRecoverableException(e, eventsToDelete);
+      }
```

Comment:
Smithy `HttpOperation` handles retry on `SmithyHttpException` with a condition, not sure if you need to handle it by yourself here.

https://github.com/aws-amplify/amplify-flutter/blob/8a7ce85f292f92403078a049aa72a073a391e23c/packages/smithy/smithy/lib/src/behavior/retryer.dart#L17-L19

Reply:
Thanks, took a look at that code.  

Unsure the conditions for when retryConfig is set or not.  

What is the general HTTPException returned?  Because all non 200 status codes aren't retryable right so in this case what Http exception would be returned

Reply:
Good point here, I've created another task and will handle general error handling for Analytics in that separate PR to come.